agent/structs/service_definition.go

```diff
@@ -251,46 +120,6 @@ func (s *ServiceDefinition) CheckTypes() (checks CheckTypes, err error) {
 	return checks, nil
 }
 
-// ServiceDefinitionConnectProxy is the connect proxy config  within a service
-// registration. Note this is duplicated in config.ServiceConnectProxy and needs
-// to be kept in sync.
-type ServiceDefinitionConnectProxy struct {
```

Comment:
Is it okay to remove this entire struct, or should only the `ExecMode` and `Command` fields be removed from the struct?

Reply:
Whole thing. Its only purpose is the `Connect.Proxy` stanza which was was purely managed proxy syntax.

Reply:
Wait... I think I'm 100% wrong.

Reply:
I'm unclear on how or if this (and the referenced `config.ServiceConnectProxy`) relates to https://github.com/hashicorp/consul/blob/856090e8934ad367c22308cbc15d06be4de1596a/api/agent.go#L109-L135